### ISSUE TITLE
cannot restart statefulset with PVC: Multi-Attach error

### ISSUE THREAD

**Issue by umegaya** [Azure/AKS#2558]
**Describe the bug**
we have aks that has 1 nodepool of 3 node, to run cockroachdb by using helm chart https://github.com/cockroachdb/helm-charts/tree/9a44d3cf217288c456fdc0c9c539ea184fec4ee3/cockroachdb
the chart creates 1 statefulset which size is same as nodepool size(3) and 1 pvc per pod (for persisting database files)

it runs nearly 1 year without any issues, but today I tried to restart 1 pod which runs on node `aks-agentpool-41524127-vmss000002` by `kubectl delete po`, then error like following (kubelet log) happens and pod never finished PodInitializing state.

```
Sep 22 02:37:00 aks-agentpool-41524127-vmss000002 kubelet[3807]: E0922 02:37:00.562916    3807 nestedpendingoperations.go:301] Operation for "{volumeName:kubernetes.io/azure-disk//subscriptions/**(snip)**/resourceGroups/mc_rg-stage-tblocksign-aks_aks-stage-jpa_japaneast/providers/Microsoft.Compute/disks/kubernetes-dynamic-pvc-6981b940-d03c-47b4-beda-8b817e95a3c6 podName: nodeName:}" failed. No retries permitted until 2021-09-22 02:39:02.562874112 +0000 UTC m=+29692320.068343912 (durationBeforeRetry 2m2s). Error: "Volume not attached according to node status for volume \"pvc-6981b940-d03c-47b4-beda-8b817e95a3c6\" (UniqueName: \"kubernetes.io/azure-disk//subscriptions/**(snip)**/resourceGroups/mc_rg-stage-tblocksign-aks_aks-stage-jpa_japaneast/providers/Microsoft.Compute/disks/kubernetes-dynamic-pvc-6981b940-d03c-47b4-beda-8b817e95a3c6\") pod \"my-cockroachdb-0\" (UID: \"c98cd5f4-8125-472e-b01a-5d01225f9627\") 
```

we tried various way to fix the problem but no hope. so finally we decided to add new node by scaling nodepool and scheduling  my-cockroachdb-0 to the new node by `kubectl drain`, then got following error. 

```
 Warning  FailedAttachVolume  27m                  attachdetach-controller  Multi-Attach error for volume "pvc-6981b940-d03c-47b4-beda-8b817e95a3c6" Volume is already exclusively attached to one node and can't be attached to another
```

so we stop `aks-agentpool-41524127-vmss000002` because we believe it **exclusively attached to the pvc**, but after that still same error happens. 

we confirmed now `pvc-6981b940-d03c-47b4-beda-8b817e95a3c6` is not attached any VM by checking following azure portal url, https://portal.azure.com/#@**(snip)**/resource/subscriptions/**(snip)**/resourceGroups/mc_rg-stage-tblocksign-aks_aks-stage-jpa_japaneast/providers/Microsoft.Compute/disks/kubernetes-dynamic-pvc-6981b940-d03c-47b4-beda-8b817e95a3c6/overview , even we can attach the disk by using `az vmss disk attach` cli. 

any idea why it happens? or any workarounds? 

regards, 

**Steps To Reproduce**
actually, we have another similar aks which runs cockroachdb cluster too, but following step causes no problem for this aks.

1. create cockroach cluster on aks by using https://github.com/cockroachdb/helm-charts/tree/9a44d3cf217288c456fdc0c9c539ea184fec4ee3/cockroachdb
2. enjoy 1 year rest in peace
3. kubectl delete po my-cockroachdb-0

**Expected behavior**
new node can attach existing pvc correctly.

**Kubernetes version**
our kubectl client may too new for our aks.
```
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.1", GitCommit:"632ed300f2c34f6d6d15ca4cef3d3c7073412212", GitTreeState:"clean", BuildDate:"2021-08-19T15:38:26Z", GoVersion:"go1.16.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.0", GitCommit:"e19964183377d0ec2052d1f1fa930c4d7575bd50", GitTreeState:"clean", BuildDate:"2020-08-26T18:49:11Z", GoVersion:"go1.15", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by andyzhangx** [comment]
could be related to this issue: https://github.com/kubernetes-sigs/cloud-provider-azure/blob/bfc60a7659c734a293baa6d8f96ce56afa8a07b1/site/content/en/FAQ/known-issues/azuredisk.md#25-multi-attach-error, could you use workaround in the link?

**Comment by umegaya** [comment]
@andyzhangx thanks! 

we used statefulset and attached different pvc for each pods, so that does not seems to be the case. 

we found actual problem is, our service principal for the aks expired a few days ago, detach on deleting pod was failed. 

I think aks should keep consistency with this kind of failure, but it is not big problem because current aks uses managed id for service principal by default, no expiration is possible. 

so I will close the issue now.
